packages/react-icons/src/utils/createSvgIcon.ts

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+// @ts-ignore
+import { css, getNativeProps, htmlElementProperties } from '@uifabric/utilities/src';
+import { classNames, MS_ICON } from 'office-ui-fabric-react/src/components/Icon/SvgIcon.styles';
+import { ISvgIconProps } from 'office-ui-fabric-react/src/components/Icon/SvgIcon.types';
+
+export type SvgIconFuncArg<TProps = ISvgIconProps> = {
+  classNames: { [iconSlot: string]: string }; // renamed from classes
+  // rtl: boolean; // how do we support this?
+  props: TProps;
+};
+
+export type SvgIconChildrenFn<TProps = ISvgIconProps> = (svgIcon: SvgIconFuncArg<TProps>) => React.ReactNode;
+
+export type SvgIconCreateFnParams<TProps> = {
+  svg: SvgIconChildrenFn<TProps & ISvgIconProps>;
+  displayName: string;
+  // handledProps?: (keyof TProps)[];
+};
+
+// TODO: SvgIconProps are different...
+const createSvgIcon = <TProps = {}>({ svg, displayName }: SvgIconCreateFnParams<TProps>) => {
+  // @ts-ignore
```

Comment:
Can you do something else besides ts-ignore here too? Like a cast to any and/or the appropriate type?

Reply:
Sry, this is still work in progress and in early prototype phase, just wanted to get more feedback on general stuff like how the factory's API, how we style things inside the svgs etc.. Will be fixed in the final stage.

Reply:
Removed